ImprovedAnvil/lib/core/animations.tph

```diff
@@ -0,0 +1,255 @@
+			PATCH_IF (~%name%~ STR_EQ ~%idx%~) BEGIN
+				LPF TO_HEX_NUMBER INT_VAR value = idx minDigits = 4 RET hexNumber END
+				PATCH_IF (NOT FILE_EXISTS_IN_GAME ~%hexNumber%.ini~ AND 
+						  ~%slotList%~ STRING_CONTAINS_REGEXP ~"%hexNumber%"~ != 0) BEGIN
+					SET slot = idx
+					SET idx = slotMax
+				END
+			END
+		END
+	END
+END
+
+// Action version of FIND_FREE_ANIM_SLOT
+DEFINE_ACTION_FUNCTION FIND_FREE_ANIM_SLOT
+INT_VAR
+	slotMin   = 0
+	slotMax   = (slotMin BAND 0xf000) + 0x1000
+	slotSteps = 1
+RET
+	slot
+BEGIN
+	OUTER_PATCH ~~ BEGIN
+		LPF FIND_FREE_ANIM_SLOT
+		INT_VAR
+			slotMin = slotMin
+			slotMax = slotMax
+			slotSteps = slotSteps
+		RET
+			slot
+		END
+	END
+END
+
+/*
+ * ===================================================
+ * ========= INDIVIDUAL ANIMATIONS ===================
+ * ===================================================
+ */
+
+// Animated Plate, Bronze (SoD/IWD)
+// uses IWD slot 0xE240
+// INI included in BG2EE
+// don't overwrite assets if already included by other mods (IAv6, EET)
+
+ACTION_IF (NOT FILE_EXISTS_IN_GAME ~mania1.BAM~) BEGIN
+	COPY ~ImprovedAnvil\resources\animations\animated_plate_bronze~ ~override~
+END
+
+// Translucent variant, because previous versions of IA garbled the Ghost slot for this monster with it's translucent=1 setting
+// uses custom INI not included in BG2EE
+// uses IWD slot allocated dynamically
+// requires patching during mod install because of dynamic allocation 
+// WRITE_LONG 0x28 (IDS_OF_SYMBOL (~animate~ ~ANVIL_ANIMATED_PLATE_BRONZE~))
+
+LAF FIND_FREE_ANIM_SLOT INT_VAR slotMin = 0xe000 slotMax = 0xefff RET slot END
+ACTION_IF (slot <= 0) BEGIN
+  FAIL ~No free creature animation slot found.~
+END
+
+LAF TO_HEX_NUMBER INT_VAR value = slot minDigits = 4 RET hexNumber END
+APPEND ~animate.ids~ ~0x%hexNumber% ANVIL_ANIMATED_PLATE_BRONZE~ UNLESS ~ANVIL_ANIMATED_PLATE_BRONZE~
+CLEAR_IDS_MAP
+
+COPY ~ImprovedAnvil\resources\animations\ini\anvil_animated_plate_bronze.ini~ ~override\%hexNumber%.ini~
+
+// Animated Plate, Dark (SoD/IWD)
+// uses IWD slot 0xE241
+// INI included in BG2EE
+// don't overwrite assets if already included by other mods (IAv6, EET)
+
+ACTION_IF (NOT FILE_EXISTS_IN_GAME ~man2a1.BAM~) BEGIN
+	COPY ~ImprovedAnvil\resources\animations\animated_plate_dark~ ~override~
+END
+
+// Animated Plate, Green (SoD/IWD)
+// uses IWD slot 0xE242
+// INI included in BG2EE
+// don't overwrite assets if already included by other mods (IAv6, EET)
+
+ACTION_IF (NOT FILE_EXISTS_IN_GAME ~man3a1.BAM~) BEGIN
+	COPY ~ImprovedAnvil\resources\animations\animated_plate_green~ ~override~
+END
+
+// Bonebat (SoD)
+// uses IWD slot 0xE294
+// INI included in BG2EE
+// don't overwrite assets if already included by other mods (IAv6, EET)
+
+ACTION_IF (NOT FILE_EXISTS_IN_GAME ~mbona1.BAM~) BEGIN
+	COPY ~ImprovedAnvil\resources\animations\bonebat~ ~override~
+END
+
+// Ghoul Queen, Acaste (converted from PST to an IWD slot) (Cuv-conversion? Extended Animations?)
```

Comment:
I'm not sure where this was sourced from. This a guess. Cuv converted PST animations to the IWD format at one point, but his pack kept the native PST scale (which si bigger than the other IE scales.)

IIRTC Extended Animations included downsamples to the common Infinity Engine scale.

Reply:
From what I remember, I extracted the original animation BAMs and used some program, BAMresizer, or something, to scale them.

Reply:
Huh, I need to doublecheck this then, because the PST bam and frame structure differs from all the other setupsd ands this might be still wrong.

Reply:
Doublechecked, this still looks correct for the IWD animation slot.